modules/wallet/update.go

```diff
@@ -453,6 +456,87 @@ func (w *Wallet) ProcessConsensusChange(cc modules.ConsensusChange) {
 	}
 }
 
+// isRelevantTxn checks if a txn is relevant to the Wallet
+func (w *Wallet) isRelevantTxn(txn types.Transaction) (relevant bool) {
+	// determine whether transaction is relevant to the wallet
+	for _, sci := range txn.SiacoinInputs {
+		relevant = relevant || w.isWalletAddress(sci.UnlockConditions.UnlockHash())
+	}
+	for _, sco := range txn.SiacoinOutputs {
+		relevant = relevant || w.isWalletAddress(sco.UnlockHash)
+	}
+	return
+}
+
+// rebroadcastOldTransaction rebroadcasts transactions that haven't been
+// confirmed within rebroadcastInterval blocks
+func (w *Wallet) rebroadcastOldTransactions(tx *bolt.Tx, cc modules.ConsensusChange) error {
+	// Get the current consensus height
+	consensusHeight, err := dbGetConsensusHeight(tx)
+	if err != nil {
+		return err
+	}
+
+	// Mark reverted transactions as not confirmed
+	for _, block := range cc.RevertedBlocks {
+		for _, bts := range w.broadcastedTSets {
+			for _, txn := range block.Transactions {
+				if _, exists := bts.confirmedTxn[txn.ID()]; exists {
+					bts.confirmedTxn[txn.ID()] = false
+				}
+			}
+		}
+	}
+
+	// Mark applied transactions as confirmed
+	for _, block := range cc.AppliedBlocks {
+		for _, bts := range w.broadcastedTSets {
+			for _, txn := range block.Transactions {
+				if _, exists := bts.confirmedTxn[txn.ID()]; exists {
+					bts.confirmedTxn[txn.ID()] = true
+				}
+			}
+		}
+	}
+
+	// Check if all transactions of the set are confirmed
+	for tSetID, bts := range w.broadcastedTSets {
```

Comment:
Are we careful to make sure the size of `w.broadcastedTSets` is relatively limited? This loop would get pretty slow if you are storing too many transactions. Practically speaking, we probably only need to keep the transactions from the last 6 blocks or so, if we have a reorg deeper than 6 blocks you probably don't want to be automatically rebroadcasting those transactions anyway - something may have changed.